src/microbit/model/image.py

```diff
@@ -0,0 +1,289 @@
+        if not isinstance(other, Image):
+            raise TypeError(
+                CONSTANTS.UNSUPPORTED_ADD_TYPE + f"'{type(self)}', '{type(other)}'"
+            )
+        elif not (other.height() == self.height() and other.width() == self.width()):
+            raise ValueError(CONSTANTS.SAME_SIZE_ERR)
+        else:
+            res = Image(self.width(), self.height())
+
+            for y in range(0, self.height()):
+                for x in range(0, self.width()):
+                    sum_value = other.get_pixel(x, y) + self.get_pixel(x, y)
+                    display_result = self.__limit_result(9, sum_value)
+                    res.set_pixel(x, y, display_result)
+
+            return res
+
+    # This multiplies image by number (if other factor is not a number, it throws an error).
+    def __mul__(self, other):
+        try:
+            float_val = float(other)
+        except TypeError:
+            raise TypeError(f"can't convert {type(other)} to float")
+
+        res = Image(self.width(), self.height())
+
+        for y in range(0, self.height()):
+            for x in range(0, self.width()):
+                product = self.get_pixel(x, y) * float_val
+                res.set_pixel(x, y, self.__limit_result(9, product))
+
+        return res
+
+    # HELPER FUNCTIONS
+
+    # This create 2D array of off LEDs with
+    # width w and height h
+    def __create_leds(self, w, h):
+        arr = []
+        for _ in range(0, h):
+            sub_arr = []
+            for _ in range(0, w):
+                sub_arr.append(0)
+            arr.append(sub_arr)
+
+        return arr
+
+    # This turns byte array to 2D array for LED field.
+    def __bytes_to_array(self, width, height, byte_arr):
+        bytes_translated = bytes(byte_arr)
+
+        if not (len(bytes_translated)) == height * width:
+            raise ValueError(CONSTANTS.INCORR_IMAGE_SIZE)
+
+        arr = []
+        sub_arr = []
+
+        for index, elem in enumerate(bytes_translated):
+            if index % width == 0 and index != 0:
+                arr.append(sub_arr)
+                sub_arr = []
+
+            sub_arr.append(elem)
+
+        arr.append(sub_arr)
+        return arr
+
+    # This converts string (with different rows separated by ":")
+    # to 2d array arrangement.
+    def __string_to_array(self, pattern):
+        initial_array, max_subarray_len = self.__string_to_initial_array(pattern)
+
+        # Fill in empty spaces in w x h matrix.
+        for arr_y in initial_array:
+            num_extra_spaces = max_subarray_len - len(arr_y)
+            for _ in range(num_extra_spaces):
+                arr_y.append(0)
+
+        return initial_array
+
+    def __string_to_initial_array(self, pattern):
+        # The result may have spaces in the 2D array
+        # and may unevent sub-array lengths
+        arr = []
+        sub_arr = []
+
+        max_subarray_len = 0
+
+        for elem in pattern:
+            if elem == ":":

```

Comment:
Awesome stuff. We should also add the ability to separate with '\n' as seen here:
https://github.com/bbcmicrobit/micropython/blob/master/docs/image.rst